src/core/tests/Test_AptitudePackageManager.py

```diff
@@ -588,6 +588,58 @@ def test_eula_not_accepted_for_patches(self):
         self.assertTrue("ACCEPT_EULA=Y" not in package_manager_for_test.single_package_dependency_resolution_template)
         self.assertTrue("ACCEPT_EULA=Y" not in package_manager_for_test.single_package_upgrade_cmd)
 
+    def test_maxpatchpublishdate_mitigation_mode(self):
+        self.runtime.set_legacy_test_type('HappyPath')
+        package_manager = self.container.get('package_manager')
+        self.assertIsNotNone(package_manager)
+        self.runtime.stop()
+
+        # classic happy path mode
+        argument_composer = ArgumentComposer()
+        argument_composer.classifications_to_include = [Constants.PackageClassification.CRITICAL]
+        argument_composer.patches_to_include = ["AzGPS_Mitigation_Mode_No_SLA", "MaxPatchPublishDate=20250101T010203Z", "*kernel*"]
+        self.runtime = RuntimeCompositor(argument_composer.get_composed_arguments(), True, Constants.APT)
+        execution_config = self.runtime.container.get('execution_config')
+        self.assertEqual(execution_config.max_patch_publish_date, "20250101T010203Z")
+        self.assertEqual(len(execution_config.included_package_name_mask_list), 1) # inclusion list is sanitized
+        self.runtime.stop()
+
+        # retains valid inclusions while honoring mitigation mode entries
+        argument_composer = ArgumentComposer()
+        argument_composer.patches_to_include = ["*kernel*", "MaxPatchPublishDate=20250101T010203Z", "AzGPS_Mitigation_Mode_No_SLA"]
+        self.runtime = RuntimeCompositor(argument_composer.get_composed_arguments(), True, Constants.APT)
+        execution_config = self.runtime.container.get('execution_config')
+        self.assertEqual(execution_config.max_patch_publish_date, "20250101T010203Z")
+        self.assertEqual(len(execution_config.included_package_name_mask_list), 1)  # inclusion list is sanitized
+        self.runtime.stop()
+
+        # missing required disclaimer entry
+        argument_composer = ArgumentComposer()
+        argument_composer.patches_to_include = ["MaxPatchPublishDate=20250101T010203Z", "*firefox=1.1"]
+        self.runtime = RuntimeCompositor(argument_composer.get_composed_arguments(), True, Constants.APT)
+        execution_config = self.runtime.container.get('execution_config')
+        self.assertEqual(execution_config.max_patch_publish_date, "")   # because no mitigation mode
+        self.assertEqual(len(execution_config.included_package_name_mask_list), 2)  # addition is ignored for removal
```

Comment:
We should have tests on what will happen during patching i.e. install, with an un-sanitized inclusion list

Reply:
Already covered - the unsanitized list will match nothing because of the specificity of the statements.